builtin/providers/aws/resource_aws_dms_replication_task.go

```diff
@@ -0,0 +1,369 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/private/waiter"
+	dms "github.com/aws/aws-sdk-go/service/databasemigrationservice"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsDmsReplicationTask() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDmsReplicationTaskCreate,
+		Read:   resourceAwsDmsReplicationTaskRead,
+		Update: resourceAwsDmsReplicationTaskUpdate,
+		Delete: resourceAwsDmsReplicationTaskDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cdc_start_time": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				// Unix timestamp 1484346880
+			},
+			"migration_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"full-load",
+					"cdc",
+					"full-load-and-cdc",
+				}, false),
+			},
+			"replication_instance_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"replication_task_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"replication_task_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replication_task_settings": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateJsonString,
+			},
+			"source_endpoint_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"table_mappings": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateJsonString,
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"target_endpoint_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsDmsReplicationTaskCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.CreateReplicationTaskInput{
```

Comment:
Looks like we still have an issue of passing empty strings - is this ok on the AWS side? We usually follow the pattern of if v_, ok := d.GetOk....